### PR TITLE
Add command line parsing to the client, minor fixes

### PR DESCRIPTION
**Pull request by icedevml**

### CODE REVIEW

File: src/Client.cpp [hunk 1]
```diff
@@ -153,50 +177,62 @@ int UrsaClient::start() {
...
     spdlog::info(
-        "    server_addr - server connection string, default: "
+        "    [server_addr]      server connection string, default: "
         "tcp://localhost:9281");
     spdlog::info(
-        "    db_command - specific command to be run in the database, "
+        "    [-q <db_command>]  specific command to be run in the database, "
         "if not provided - interactive mode");
+    spdlog::info(
+        "    [-s]               silent mode, dump only command output");
```
Comment (chivay) [inline src/Client.cpp:200]: I would reconsider argument switches:
`-q` - quiet
`-c` - command

File: src/Client.cpp [hunk 2]
```diff
@@ -153,50 +177,62 @@ int UrsaClient::start() {
         }
     }
 
-    // wait for status thread to terminate
-    status_th.join();
+    status_th.detach();
```
Comment (chivay) [inline src/Client.cpp:180]: It seems wrong to detach the thread

File: src/Client.cpp [hunk 3]
```diff
@@ -153,50 +177,62 @@ int UrsaClient::start() {
...
+                silent_mode = true;
+                break;
+            case 'q':
+                db_command = optarg;
+                break;
+            case 'h':
+                print_usage(argc >= 1 ? argv[0] : "ursacli");
+                return 0;
+            default:
+                print_usage(argc >= 1 ? argv[0] : "ursacli");
```
Comment (chivay) [inline src/Client.cpp:222]: I think we can assume that argc is >= 1. If someone does cursed things, they're on their own.

File: src/Client.cpp [hunk 4]
```diff
@@ -8,28 +10,36 @@
 #include "libursa/ZHelpers.h"
 #include "spdlog/spdlog.h"
 
-static void wait_sec() {
-    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+bool UrsaClient::wait_sec() {
```
Comment (msm-code) [inline src/Client.cpp:13]: This looks pretty hacky, but I don't know why your threads kept hanging.

Just in case, are you sure this solves the problem, or just makes it less visible? (i.e. now the time window for a bug is 10ms instead of 1000ms).

Also, how CPU intensive is this? If ursacli uses something like 10% cpu by nooping it'd be pretty bad.

File: src/Client.cpp [hunk 5]
```diff
@@ -98,10 +113,14 @@ void UrsaClient::recv_res(zmq::socket_t &socket) {
         this->command_active = false;
         auto res = json::parse(res_str);
 
-        if (res["type"] == "error") {
-            spdlog::error(res["error"]["message"].get<std::string>());
+        if (!this->silent_mode) {
+            if (res["type"] == "error") {
+                spdlog::error(res["error"]["message"].get<std::string>());
+            } else {
+                spdlog::info(res.dump(4));
```
Comment (msm-code) [inline src/Client.cpp:120]: I'd change this to cout too (i.e. always print results to the stdout)

File: src/Client.cpp (hunk 1 above)
Comment (msm-code) [reply src/Client.cpp:200]: :+1:

File: src/Client.cpp (hunk 3 above)
Comment (msm-code) [reply src/Client.cpp:222]: What if someone starts ursacli from their shellcode using execve with an empty argv array?

Comment (chivay) [reply src/Client.cpp:222]: :duck:

File: src/Client.cpp (hunk 2 above)
Comment (icedevml) [reply src/Client.cpp:180]: idk why but when doing `join()` there is a random hang from time to time, not able to reproduce on `Debug` build :| do you have any clue?

File: src/Client.cpp (hunk 4 above)
Comment (icedevml) [reply src/Client.cpp:13]: it's not related to any bug, imagine that you run
```
ursacli -c status
```
and the status thread loops for 1 second, so you get the output immediately but the program is waiting additional 1 second before the status thread realizes that it should terminate

Comment (msm-code) [reply src/Client.cpp:13]: How did we solve this in python? I don't recall any threads there.

IIRC we just had a while() loop waiting for response with 1s timeout, and printing a progress bar every 1s. Maybe this will solve all these concurrency and threading problems?

Comment (icedevml) [reply src/Client.cpp:13]: Then we should probably do some short-time polling.

Comment (msm-code) [reply src/Client.cpp:13]: I mean, in this case it works like this:

```
while True:
    x = recv(timeout=1s)
    if timeout():
        print_progress()
        continue
    else:
        print_result(x)
        break
```

File: src/Client.cpp (hunk 2 above)
Comment (icedevml) [reply src/Client.cpp:180]: already resolved, there was a data race

File: src/Client.cpp (hunk 4 above)
Comment (icedevml) [reply src/Client.cpp:13]: I would skip async pooling for now, as it would require a huge rework of the client (introducing new bugs), while at the moment we've (with @chivay) realized how to patch the existing ones :P

File: src/Client.cpp (hunk 3 above)
Comment (icedevml) [reply src/Client.cpp:218]: idk, this doesn't cost anything bug we have strong assertion of not going out of bounds here
